### DISCUSSION TITLE
ESP-IDF example code available?

### DISCUSSION THREAD

**Discussion by parkykk** [lovyan03/LovyanGFX#241] [Q&A]
Hi Lovyan03 

Would you be able to provide ESP-IDF example code for MakerabsParallelTFTwithTouch ? 

kind regards

**Comment by lovyan03** [comment]
I recently added an example configuration.

https://github.com/lovyan03/LovyanGFX/blob/master/src/lgfx_user/LGFX_ESP32S2_MakerabsParallelTFTwithTouch.hpp

If you include this and create an instance of LGFX, I believe you can use it.

**Reply by parkykk** [reply]
Hi lovyan03 

Thanks for your prompt reply. 

I am a newbie to programming. I am just testing compile only at the moment. However it prompts compile error with the below code. 

Could you please advise how I can fix it?

I am using ESP-IDF with VScode. I copied all files in **LovyanGFX-master\src** under main directory(project file). 

I tried to compile the code but it prompts errors.

-------------------------------------------------------------------------------------------
#define LGFX_USE_V1

#include "lgfx_user/LGFX_ESP32S2_MakerabsParallelTFTwithTouch.hpp"

#define LCD_CS 37
#define LCD_BLK 45

**static LGFX lcd;**  

extern "C" void app_main(void)
{
   //
}
-----------------------------------------------------------------------------------------------

If I comment out the line ### static LGFX lcd; (or LGFX lcd;), it compiles successfully, but it generates compile error the other way around. 

Kind regards 
parkykk

**Reply by lovyan03** [reply]
Please refer to this Issue.
https://github.com/lovyan03/LovyanGFX/issues/173

Find out and figure out how to use the components.
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/build-system.html#example-project
